### PR TITLE
 Add node and build queue metrics.

### PR DESCRIPTION
**Pull request by mst-ableton**
ptal!

### CODE REVIEW

File: jenkins_exporter.py [hunk 1]
```diff
@@ -22,8 +22,9 @@ class JenkinsCollector(object):
 
     def __init__(self, target, user, password):
         self._target = target.rstrip("/")
-        self._user = user
-        self._password = password
+        self._auth = None
+        if user is not None and password is not None:
```
Comment (nre-ableton) [inline jenkins_exporter.py]: Might this be replaced with `if user and password:`?

File: jenkins_exporter.py [hunk 2]
```diff
@@ -56,10 +57,17 @@ def _request_data(self):
         }
 
         def parsejobs(myurl):
-            # params = tree: jobs[name,lastBuild[number,timestamp,duration,actions[queuingDurationMillis...
-            response = requests.get(myurl, params=params, auth=(self._user, self._password))
+            now = time.time()
```
Comment (nre-ableton) [inline jenkins_exporter.py]: `now` won't be now when you use it later. :) How about `initial_time` or something like that?

File: jenkins_exporter.py [hunk 3]
```diff
@@ -79,6 +83,105 @@ def parsejobs(myurl):
 
         return parsejobs(url)
 
+    def _request_queue(self, jobs):
+        url = '{0}/queue/api/json'.format(
```
Comment (nre-ableton) [inline jenkins_exporter.py]: Why is this statement broken on multiple lines?

File: jenkins_exporter.py [hunk 4]
```diff
@@ -79,6 +83,105 @@ def parsejobs(myurl):
...
+                    [job, config], task_count[job][config])
+
+    def _request_nodes(self):
+        url = '{0}/computer/api/json'.format(self._target)
+        tree = 'computer[displayName,offline,temporarilyOffline,idle,monitorData[*]]'
+        params = {
+            'tree': tree,
+        }
+
+        now = time.time()
```
Comment (nre-ableton) [inline jenkins_exporter.py]: Same comment on variable naming here

File: jenkins_exporter.py (hunk 1 above)
Comment (mst-ableton) [reply jenkins_exporter.py]: *blinks* yeah...

**Comment by mst-ableton** [comment]
@nre-ableton pt(another)l!

**Comment by mst-ableton** [comment]
@ablbot autosquash

**Pull request merged**